modules/fretboard/index.js

```diff
@@ -71,3 +71,50 @@ export function scale (tuning, scale, first, last) {
   var filter = map(scaleFilter(scale))
   return build(tuning, first, last).map(filter)
 }
+
+/**
+ * Build an array of reachable chord shapes based on given notes and tuning.
+ * @param {String|Array} tuning - the tuning name or notes
+ * @param {Array} notes - an array of chord notes
+ * @param {Integer} first - the first fret number
+ * @param {Integer} last - the last fret number
+ * @return {Array} An array of arrays, one for each possible shape.  Index is string index [ 'E2', 'B2', 'E3', 'G#3', 'B3', 'E4' ]
+ */
+export function chordShapes (tuning, notes, first, last, span) {
+  var chordNotes = scale(tuning, notes, first, last)
+  var chordGroups = []
+
+  // Break each string array into {fretSpan} frets overlapping sections
+  var strings = chordNotes.map(function(v, i) {
+    return v.map(function(v2, i2) {
+      return v.slice(i2, i2 + span).map(function(v3, i3) {
+        // Convert note names to fret numbers
+        return v3 !== null ? i2 + i3  : null
+      })
+    })
+  })
+
+  // Build chordGroups
+  strings.forEach(function(v, i) {
+    v.forEach(function(v2, i2) {
+      if (!Array.isArray(chordGroups[i2])) chordGroups[i2] = []
+
+      // Strip null values
+      var frets = v2.filter(function(v3) {
+        return v3 !== null
+      })
+
+      if (frets.length <= 1) {
+        chordGroups[i2].push(frets.toString() ? frets.toString() : null)
+      } else {
+        chordGroups[i2].push(frets)
+      }
+    })
+  })
+
+  // Remove null and neighboring duplicate arrays
+  return chordGroups.filter(function(v, i) {
+    if (!v.join('').toString()) return false;
```

Comment:
I would rewrite that snippet like: `...filter(function(v) { return compact(v).length })` 

For me it's clear enough... What do you think?